mlir/include/mlir/IR/Operation.h

```diff
@@ -500,6 +500,12 @@ class alignas(8) Operation final
                                llvm::to_vector(getDiscardableAttrs()));
   }
 
+  /// Return all attributes that are not stored as properties.
+  DictionaryAttr getNonPropertyAttrDictionary() {
```

Comment:
I have concerns about this kind of API: usage of properties should be something abstracted, there is something wrong if we leak this now.

Reply:
I don't like it much either. However, we got into a situation that conflates inherent/discardable attributes and properties. In particular, a lot of downstreams that haven't migrated to properties, as well as some pieces of upstream, incorrectly assume that all attributes are discardable because `getDiscardableAttrs` used to return inherent attributes when properties were not used.